library/ssl_tls12_client.c

```diff
@@ -2034,13 +2034,11 @@ static int ssl_get_ecdh_params_from_cert(mbedtls_ssl_context *ssl)
     ssl->handshake->ecdh_psa_type = PSA_KEY_TYPE_ECC_KEY_PAIR(ecc_family);
 
     /* Store peer's public key in psa format. */
-    ret = mbedtls_ecp_point_write_binary(&peer_key->grp, &peer_key->Q,
-                                         MBEDTLS_ECP_PF_UNCOMPRESSED, &olen,
-                                         ssl->handshake->ecdh_psa_peerkey,
-                                         MBEDTLS_PSA_MAX_EC_PUBKEY_LENGTH);
+    ret = mbedtls_pk_get_public_key(peer_pk, ssl->handshake->ecdh_psa_peerkey,
```

Comment:
Note: this is something we _want_ to do at some point (it's part of #7405) but I'm concerned that it we _have_ to do it now, it's a compatibility smell. So, is this something you did because you could, of because you had to?

Reply:
As for the [previous comment](https://github.com/Mbed-TLS/mbedtls/pull/7202#discussion_r1165432351) I suspect that all these cases in which I forcedly have to recover some "alignment" between the legacy mechanism and the new one is related to the period in which this new `USE_PSA` mechanism was a replacement for the old one.